extensions/community/PokiGamesSDK.json

```diff
@@ -0,0 +1,321 @@
+  "shortDescription": "Extension that implements Poki SDK so that your game can be hosted on Poki.",
+  "version": "0.0.1",
+  "description": [
+    "Please refer to [Poki's guide](https://sdk.poki.com/sdk-documentation.html) to know when to call the different actions.",
+    "",
+    "To initialize the SDK, use the asynchronous action \"Load SDK\" at the beginning of your very first scene."
+  ],
+  "tags": [
+    "poki",
+    "ads",
+    "sdk"
+  ],
+  "authorIds": [
+    "tgIIofwSWVNtHsFW7AR2imu8gCj1"
+  ],
+  "dependencies": [],
+  "eventsFunctions": [
+    {
+      "fullName": "",
+      "functionType": "Action",
+      "name": "onFirstSceneLoaded",
+      "sentence": "",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": "gdjs._pokiGamesSdkExtension = {};",
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "async": true,
+      "description": "Load Poki SDK.",
+      "fullName": "Load SDK",
+      "functionType": "Action",
+      "group": "Poki Games",
+      "name": "LoadSDK",
+      "sentence": "Load Poki SDK",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "function addScript(src) {",
+            "    return new Promise((resolve, reject) => {",
+            "        const scriptElement = document.createElement('script');",
+            "",
+            "        scriptElement.setAttribute('src', src);",
+            "        scriptElement.addEventListener('load', resolve);",
+            "        scriptElement.addEventListener('error', reject);",
+            "",
+            "        document.body.appendChild(scriptElement);",
+            "    });",
+            "}",
+            "addScript('https://game-cdn.poki.com/scripts/v2/poki-sdk.js').then(() => {",
+            "    PokiSDK.init().then(() => {",
+            "        console.log(\"Poki SDK successfully initialized\");",
+            "        gdjs.evtTools.common.resolveAsyncEventsFunction(eventsFunctionContext);",
+            "    }).catch(() => {",
+            "        console.log(\"Initialized, but the user likely has adblock\");",
+            "    });",
+            "})"
+          ],
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": true
+        }
+      ],
+      "parameters": [],
+      "objectGroups": []
+    },
+    {
+      "description": "Inform Poki game finished loading.",
+      "fullName": "Game loading finished",
+      "functionType": "Action",
+      "group": "Poki Games",
+      "name": "CallGameLoadingFinished",
+      "sentence": "Inform Poki game finished loading",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "PokiSDK.gameLoadingFinished();",
```

Comment:
If you only conditionally load the Poki SDK, you cannot just call it like this, since it is not guaranteed to exist, and calling a function that does not exist will crash the game. Unless you always load Poki when the first scene loads and do so directly without a DOM script so that the game is guaranteed to run with the Poki SDK loaded at all times, prefix all poki sdk calls with `if (typeof PokiSDK !== "undefined") `.